### DISCUSSION TITLE
LINQ .NET

### DISCUSSION THREAD

**Discussion by RamonesCode** [balta-io/2801#105] [Perguntas e Respostas] [answered]
![LINQ](https://user-images.githubusercontent.com/84512227/186510331-b9ebf464-d5ca-4690-8570-9b31b4442de4.png)

Boa tarde professor,  obrigado pelo seu tempo desde já , 

Vi recentemente o vídeo que publicou sobre algumas melhorias no .net6  e hoje trabalhando me deparei com um problema que eu precisava fazer uma negação com o Linq , e pensei puxa será que há algo do tipo "not equal" disponível , e percebi nos google que fiz que não há exatamente isso , mas maneiras paliativas disso. 
Com isso vem minha pergunta, o senhor sabe se haverá alguma mudança ou se há melhorias previstas para o linq ? Pergunto  não só por esse acontecimento , mas porque tenho usado muito arrow function e percebo que é muito mais comum do que o linq.

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde @RamonesCode , como vai?

O LINQ está sempre recebendo melhorias, mas não sei dizer se tem algo previsto para isto... na verdade não entendi muito bem o contexto e o que precisa fazer, se puder detalhar mais as entidades e o que precisa consultar, talvez eu tenha uma opinião diferente.

**Comment by RamonesCode** [comment]
Bom dia @andrebaltieri , bem graças a Deus obrigado, e obg pelo retorno.

Show ! Entendi. 

Explicando um pouco o contexto, eu estava trabalhando em uma integração e precisava realizar uma validação em um JOIN  de uma lista de dados coletados (areaList) e uma lista de dados que eu já tinha em bd (cameraAreaList), eu queria validar dois itens de ambas as listas , mas de forma que eles deveria diferir e aí que pensei na existência de um "not equal" . O resultado deveria me retornar um nova lista (listExpira) contendo apenas os itens que existissem no meu bd mas que houvesse sido modificado [o campo dsCoordinates] ( a lista cameraAreaList é um armazenamento local da areaList que por sua vez é o armazenamento aws  onde  os dados da integração com o equipamento fica). 
Mas acabei fazendo de um outro jeito, conforme a foto , e "funfou".  

Se eu conseguir explanar um pouquinho e o senhor tiver compreendido e tiver uma outra ótica eu fico feliz em ouvir . 

Muito obrigado 
![example2](https://user-images.githubusercontent.com/84512227/187224548-f145cd60-1eab-4835-b9d2-1474ec39d4f9.png)

**Reply by gilivasconcelos** [reply]
Olá, já tive essa dúvida tempo atrás, oque encontrei na época fi o o comando Except, você usa para trazer dados que estão em uma lista de dados "A" mas não estão na lista de dados "B" seria como se fosse a diferença na teoria matemática do conjuntos, então é algo que eu acho que serviria. No LINQ, assim como no SQL, podemos ter a união(Union ), a diferença(Except) e a intersecção (Join) ! :-)   
![image](https://user-images.githubusercontent.com/21987123/188519855-0c797055-3bd0-4a74-b388-e28bb5bab59c.png)
